File: src/main/java/at/hannibal2/skyhanni/utils/jsonobjects/LocationFixJson.java
package at.hannibal2.skyhanni.utils.jsonobjects;

import at.hannibal2.skyhanni.utils.LorenzVec;
import com.google.gson.annotations.Expose;

import java.util.Map;

public class LocationFixJson {

    @Expose
    public Map<String, LocationFix> locationFixes;

    public static class LocationFix {
        @Expose
        public LorenzVec a;

        @Expose
        public LorenzVec b;

        @Expose
        public String island_name;

        @Expose
        public String real_location;
    }

}

File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/repo/LocationFixJson.kt
package at.hannibal2.skyhanni.data.jsonobjects.repo

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import net.minecraft.world.phys.Vec3

data class LocationFixJson(
    @Expose val locationFixes: Map<String, LocationFix>,
)

data class LocationFix(
    @Expose val a: Vec3,
    @Expose val b: Vec3,
    @Expose @SerializedName("island_name") val islandName: String,
    @Expose @SerializedName("real_location") val realLocation: String,
)
